[WIP] Changes for score collector for dashboard

sidbha007 opened the pull request:
Feature to score team dashboards
- Overall score for a dashboard between 0-5
- Score for individual widget data: build, scm, quality, deploy 

Changes include
- Score collector to calculate score for enabled dashboards (collectors/misc/score-collector)
- UI widget to display and configure score
- Core models, repositories & API changes

Score Collector:
- Collector will run periodically to calculate score for enabled dashboards
- Configurations for score calculation can be set at default and individual widgets
- Configurable criteria to set score if it meets a threshold.

Flow to view score for a dashboard
1. Enable score widget by clicking on configure and save
 <img width="365" alt="score_view_config" src="https://user-images.githubusercontent.com/5722038/36446664-f32ee858-1679-11e8-8e1d-eefe7883970c.png">

2. After score collector has run, you should see overall score for the team dashboard
<img width="332" alt="score_view" src="https://user-images.githubusercontent.com/5722038/36446712-1c5c6c64-167a-11e8-8f07-fdb0f0adae5f.png">

3. Details for score can be viewed by clicking `View Details`
<img width="961" alt="score_view_details" src="https://user-images.githubusercontent.com/5722038/36446744-31407af8-167a-11e8-90d6-697ea7899db9.png">

4. Individual widgets will have score view, and clicking on it will show details of the score
<img width="693" alt="score_widget_view" src="https://user-images.githubusercontent.com/5722038/36446784-4dc0d74a-167a-11e8-9fed-7ecb2b3dcb2e.png">

sidbha007 committed:
Changes for score collector for dashboard

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/walmartlabs/Hygieia?pullRequest=1) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/walmartlabs/Hygieia?pullRequest=1) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: sidbha007<br/>:x: Tapabrata "Topo" Pal<br/><hr/>**Tapabrata "Topo" Pal** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/walmartlabs/Hygieia?pullRequest=1) it.</sub>

sidbha007 committed:
Fixes and updates
Core
-  Add ScoreValueType enum, in future if score is calculated for different types like Team, User
- Update models to have map options for additional data

Score Collector
- Fix score tests
- Fix score calculation for builds
- Remove options specific to walmart for scoring in scm and deploy
UI
- Add changes in capitalone template
- Fixes to view scoring

sidbha007 committed:
Cleanup

sidbha007 committed:
Cleanup and comments

sidbha007 committed:
Score service and controller Unit Tests

sidbha007 committed:
Comments for widget score calculations

sidbha007 committed:
Add examples for score collector settings

sidbha007 committed:
Move score css to its own `less` file

sidbha007 committed:
Merge branch 'master' of https://github.com/capitalone/Hygieia into scoreDashboard

sidbha007 committed:
Cleanup score widget

sidbha007 committed:
Merge branch 'master' of https://github.com/capitalone/Hygieia into scoreDashboard

sidbha007 committed:
Add score details ui test data

sidbha007 committed:
Merge branch 'master' of https://github.com/capitalone/Hygieia into scoreDashboard

sidbha007 committed:
Fixes
- remove unused variables and code
- remove settings type and custom config settings as it is no longer use

sidbha007 committed:
Remove unused method and call cleanup for score collector items

sidbha007 committed:
Remove unused import

sidbha007 committed:
Changes to add widgetId reference in score metrics model
- Made model names generic with Score`Component`Metric as name.
- Add refId property in ScoreComponentMetrics to have reference Ids. This would have widgetId reference when score is for dashboard
- Change `id, name` properties to `displayId, displayName`
- Use refId to show score for a widget in UI
- other cleanup

sidbha007 committed:
Add timestamp for score metric

sidbha007 committed:
Save score criteria settings changes
- mongodb collection score_criteria_settings to save settings
- refactor and move settings classes to core
- changes to save criteria settings on collector run

sidbha007 committed:
remove old classes

sidbha007 committed:
Add api to fetch score criteria settings by type

Commit:
Merge branch 'master' into scoreDashboard

sidbha007 committed:
API changes to enable score for a dashboard in create/update request
- Add properties for scoreEnabled and scoreDisplay in dashboard request.
- ScoreDashboardService to associate collectorItem for score dashboard
- Fix tests

sidbha007 committed:
Changes to enable/disable score settings
- UI changes in create/edit dashboard screen to enable/disable scoring
- API to update enable/disable scoring of dashboard

sidbha007 committed:
Use objectid for saving dashboardId in score collector item

sidbha007 committed:
Initial score display in header
- directive to display widget in header
- api to fetch score by dashboard id
- common score repository

sidbha007 committed:
Remove old files

sidbha007 committed:
UI Changes to view score in widget
- directive to view in widget
- update in templates
- show score as first item in custom/widget templates
- fixes

sidbha007 committed:
Style score in widget to show score on top for better readability

sidbha007 committed:
Add comments and UTs for ScoreDashboardService

sidbha007 committed:
Add UTs for DashboardService and DashboardController

sidbha007 committed:
Cleanup

sidbha007 committed:
Fix to show message "no details found" in score details when data is undefined

sidbha007 committed:
Add ui constants for ScoreDisplayType : HEADER, WIDGET

sidbha007 committed:
Merge branch 'master' of https://github.com/capitalone/Hygieia into scoreDashboard

sidbha007 committed:
Fix for review: revert score props from DashboardRemoteRequest, DashboardFixture and related tests

sidbha007 committed:
cleanup: remove unwanted injection of `$rootScope`
